docs/reference/mapping/types/histogram.asciidoc

```diff
@@ -0,0 +1,87 @@
+[role="xpack"]
+[testenv="basic"]
+[[histogram]]
+=== Histogram datatype
+++++
+<titleabbrev>Histogram</titleabbrev>
+++++
+
+A `histogram` field stores pre-aggregated numerical data represented as a histogram. These histograms
+are defined by a set of pairs, one being the value representing the bucket and the count of that value.
+THe values should be defined accordingly to the type of aggregation this field will be used for.
+
+Histograms can be used for aggregations, in particular <<search-aggregations-metrics-percentile-aggregation, percentiles>>
+and <<search-aggregations-metrics-percentile-rank-aggregation, percentile ranks>> aggregations are supported.
+
+Histograms are indexed by defining two arrays, one containing the `values` (or buckets) for the histogram and the other containing
+the `counts` for such values. Those two arrays must have the same length.
```

Comment:
This section is great, but I think it's missing some crucial information:

- The `values` array stores `double` floats
- The `counts` array stores `long` integers
- The field data is stored as doc values and not indexed
- Because it's not indexed, it can't be used for most queries

I also think some of this section can get repetitive and needs some chunking. I left a suggestion below, but don't feel obligated to use it.

```
A  field to store pre-aggregated numerical data representing a histogram.
This data is defined using two paired arrays:

* A `values` array of <<number,`double`>> numbers, representing the buckets for
  the histogram. These values must be provided in ascending order.
* A corresponding `counts` array of  <<number,`long`>> numbers, representing how
  many values fall into each bucket.

Because the elements in the `values` array correspond to the elements in the
same position of the `count` array, these two arrays must have the same length.

[[histogram-uses]]
==== Uses

`histogram` fields are primarily intended for use with aggregations. To make it
more readily accessible for aggregations, `histogram` field data is stored as a
binary <<doc-values,doc values>> and not indexed.

Because the data is not indexed, you only can use `histogram` fields for the
following aggregations and queries:

* <<search-aggregations-metrics-percentile-aggregation,percentiles>> aggregation
* <<search-aggregations-metrics-percentile-rank-aggregation,percentile ranks>> aggregation
* <<query-dsl-exists-query,exists>> query

We recommend you define the buckets in the `values` array based on the type of aggregation you intended to use.
```